pytorch_lightning/trainer/trainer.py

```diff
@@ -505,13 +505,15 @@ def _call_and_handle_interrupt(self, trainer_fn: Callable, *args: Any, **kwargs:
         """
         try:
             return trainer_fn(*args, **kwargs)
+        # TODO(@daniellepintz): treat KeyboardInterrupt as BaseException (delete the code below) in v1.7
         except KeyboardInterrupt as exception:
             rank_zero_warn("Detected KeyboardInterrupt, attempting graceful shutdown...")
             # user could press Ctrl+c many times... only shutdown once
             if not self.interrupted:
                 self.state.status = TrainerStatus.INTERRUPTED
                 self.on_keyboard_interrupt()
                 self.on_exception(exception)
+                raise exception
```

Comment:
@daniellepintz I would leave this change to another PR. This seemed like an intentional choice from https://github.com/PyTorchLightning/pytorch-lightning/pull/856/ to not re-raise the exception. Doing now will be a breaking change. Let's discuss this on https://github.com/PyTorchLightning/pytorch-lightning/issues/9286

Reply:
sg!